src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,137 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_ARRAYLIST_CAPACITY = 10;
+    private int size;
+    private Object[] elements;
+
+    public ArrayList() {
+        elements = new Object[DEFAULT_ARRAYLIST_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        ensureCapacity(size + 1);
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkBoundExclusive(index);
+        if (index == size || size >= elements.length) {
+            ensureCapacity(size + 1);
+        }
+        for (int i = size; i > index; i--) {
+            elements[i] = elements[i - 1];
+        }
+        elements[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (list == null) {
+            throw new NullPointerException("List can't be null");
+        }
+        int newSize = size + list.size();
+        ensureCapacity(newSize);
 
+        for (int i = 0; i < list.size(); i++) {
```

Comment:
Don't create repeating code. The null check inside the loop is unnecessary. If the list contains a null element, it should be added to 'elements' just like any other element.